pkg/cli/admin/release/extract.go

```diff
@@ -296,9 +296,100 @@ func (o *ExtractOptions) Run() error {
 		}
 	}
 
+	tarEntryCallbacks := []extract.TarEntryFunc{}
+
+	var manifestErrs []error
+	if o.ExtractManifests || o.CredentialsRequests {
+		expectedProviderSpecKind := ""
+		if len(o.Cloud) > 0 {
+			expectedProviderSpecKind = credRequestCloudProviderSpecKindMapping[o.Cloud]
+		}
+
+		tarEntryCallbacks = append(tarEntryCallbacks, func(hdr *tar.Header, _ extract.LayerInfo, r io.Reader) (bool, error) {
+			if ext := path.Ext(hdr.Name); len(ext) == 0 || !(ext == ".yaml" || ext == ".yml" || ext == ".json") {
+				return true, nil
+			}
+			klog.V(4).Infof("Found manifest %s", hdr.Name)
+			raw, err := ioutil.ReadAll(r)
+			if err != nil {
+				manifestErrs = append(manifestErrs, errors.Wrapf(err, "error reading file %s", hdr.Name))
+				return true, nil
+			}
+			ms, err := manifest.ParseManifests(bytes.NewReader(raw))
+			if err != nil {
+				manifestErrs = append(manifestErrs, errors.Wrapf(err, "error parsing %s", hdr.Name))
+				return true, nil
+			}
+			o.Manifests = append(o.Manifests, ms...)
+
+			if o.CredentialsRequests {
+				credRequestManifests := []manifest.Manifest{}
+				for _, m := range ms {
+					if m.GVK != credentialsRequestGVK {
+						continue
+					}
+					if len(expectedProviderSpecKind) > 0 {
```

Comment:
It's a string so ` != ""` would be more idiomatic

Reply:
Done with 261e686b7 -> e125e01d1.